app/helpers/report_helper/timer.rb

```diff
@@ -0,0 +1,70 @@
+ReportHelper::Timer = Struct.new(
+  :typ,
+  :months,
+  :weeks,
+  :days,
+  :hours,
+  :start_date,
+  :start_hour,
+  :start_min,
+) do
+  def initialize(typ = nil, months = 1, weeks = 1, days = 1, hours = 1, start_date = nil, start_hour = '00', start_min = '00')
+    super
+  end
```

Comment:
Yeah, I think I'd lean more toward a class here too.

It's a tough line to draw, but I'd say when you get beyond a couple of values, struct looks less appealing.

(I'm also surprised this code works... I thought the above was exactly the reason people inherit from structs instead of using the block form.)

---

I must note that a struct marshals slightly more compactly than the equivalent plain object, but the difference is trivial.